cmd/relayproxy/testdata/config/valid-file.yaml

```diff
@@ -7,3 +7,6 @@ retriever:
 exporter:
   kind: log
 enableSwagger: true
+apiKeys:
+  apikey1: true
+  apikey2: true
```

Comment:
I am not sure to understand the boolean here.
Would it be better to have only a list, something like 
```yaml
apiKeys:
  - apikey1 # with a comment explaining who is the owner of the key
  - apikey2
```

I think it will make the configuration representation way easier to understand.
You can still put it in a set in the config struct, but in terms of configuration, it will be way more understandable for the users of the relay proxy.